src/commands/bug_report.rs

```diff
@@ -0,0 +1,419 @@
+            Err(e.into())
+        }
+    }
+}
+
+fn auto_complete(p: String) -> Completions<String> {
+    use std::path::Path;
+
+    let current: &Path = p.as_ref();
+    let (mut dir, last) = if p.ends_with('/') {
+        (current, "")
+    } else {
+        let dir = current.parent().unwrap_or_else(|| "/".as_ref());
+        let last = current
+            .file_name()
+            .and_then(std::ffi::OsStr::to_str)
+            .unwrap_or("");
+        (dir, last)
+    };
+
+    if dir.to_str().unwrap().is_empty() {
+        dir = ".".as_ref();
+    }
+
+    let mut files: Completions<_> = match dir.read_dir() {
+        Ok(files) => files
+            .flatten()
+            .filter_map(|file| {
+                let path = file.path();
+                let is_dir = path.is_dir();
+                match path.into_os_string().into_string() {
+                    Ok(s) if is_dir => Some(s + "/"),
+                    Ok(s) => Some(s),
+                    Err(_) => None,
+                }
+            })
+            .collect(),
+        Err(_) => {
+            return completions![p];
+        }
+    };
+
+    if files.is_empty() {
+        return completions![p];
+    }
+
+    let fuzzer = SkimMatcherV2::default();
+    files.sort_by_cached_key(|file| fuzzer.fuzzy_match(file, last).unwrap_or(i64::MAX));
+    files
+}
+
+// ============================================================================= //
+
+struct Environment {
+    shell_info: ShellInfo,
+    os_type: os_info::Type,
+    terminal_info: TerminalInfo,
+    os_version: os_info::Version,
+}
+
+#[derive(Debug)]
+struct TerminalInfo {
+    name: String,
+    version: String,
+}
+
+fn get_terminal_info() -> TerminalInfo {
+    let terminal = std::env::var("TERM_PROGRAM")
+        .or_else(|_| std::env::var("LC_TERMINAL"))
+        .unwrap_or_else(|_| UNKNOWN_TERMINAL.to_string());
+
+    let version = std::env::var("TERM_PROGRAM_VERSION")
+        .or_else(|_| std::env::var("LC_TERMINAL_VERSION"))
+        .unwrap_or_else(|_| UNKNOWN_VERSION.to_string());
+
+    TerminalInfo {
+        name: terminal,
+        version,
+    }
+}
+
+#[derive(Debug)]
+struct ShellInfo {
+    name: String,
+    version: String,
+}
+
+fn get_shell_info() -> ShellInfo {
+    let failure_shell_info = ShellInfo {
+        name: UNKNOWN_SHELL.to_string(),
+        version: UNKNOWN_VERSION.to_string(),
+    };
+
+    let current_shell = match Shell::from_env() {
+        Some(shell) => shell.to_string(),
+        None => return failure_shell_info,
+    };
+
+    let version = get_shell_version(&current_shell);
+
+    ShellInfo {
+        version,
+        name: current_shell.to_string(),
+    }
+}
+
+fn get_shell_version(shell: &str) -> String {
+    let time_limit = Duration::from_millis(500);
+    match shell {
+        "powershell" => exec_cmd(
+            shell,
+            &["(Get-Host | Select Version | Format-Table -HideTableHeaders | Out-String).trim()"],
+            time_limit,
+        ),
```

Comment:
Do we really want to support windows?

Reply:
i think not supporting a major OS is pretty detrimental to OSS

Reply:
I haven't done any testing in Windows so I'm not even sure this tool works there. @xynydev what's your opinion on Windows support and should this be a larger discussion?

Reply:
yeah we can always postpone it. I just dont want to exclude a section of developers, but tools do it all the time so either or.

We can always say ⭐  comming soon ™️ ⭐

Reply:
Definitely with the tm lol. I personally wouldn't mind supporting Windows later, but right now we should get it working for the main user-base which would be Linux users.

Reply:
> @xynydev what's your opinion on Windows support and should this be a larger discussion?

@gmpinder maybe a larger discussion, if needed. IMO Win users would be better off using these tools under WSL anyway. They'd need container tools and git and such. I doubt that we can make Win have as good of support as Linux without a maintenance burden... Who's out there building containers and desktop Linux images on Windows anyway?